### DISCUSSION TITLE
Add support for high bit depth multichannel images

### DISCUSSION THREAD

**Discussion by aclark4life** [python-pillow/Pillow#8451] [Ideas]
We now have two approaches for this:

- https://github.com/python-pillow/Pillow/pull/8224
- https://github.com/python-pillow/Pillow/pull/8330

Discuss 🍿

**Comment by aclark4life** [comment]
If I understand correctly, the difference between the two:

- Update PIL's C data structure to accommodate high bit depth multi channel images.
- Add Arrow's C data structure library to Pillow to accommodate high bit depth multi channel images.

With the former, we have the potential for making pointer math mistakes but avoid including Arrow's C library in Pillow. With the latter we have a more robust data structure for the low price of including Arrow's C library.

**Comment by wiredfool** [comment]
That's not really correct --- My arrow support here doesn't do anything to enable high bit depth images. All it does is use the lightest weight arrow integration (headers/structs only) to export the existing pillow image memory blocks as arrow using the __arrow_c_array__` interface, which is the spiritual successor to the `__array_interface__`. There are a few caveats, in that the data has to be packed, so there are formats that won't work (anything with row padding/alignment), and it currently doesn't support multi block images (>16mb by default). 

I'd like to fix those, and get the reverse working -- read arrow array as an image.
